File: sdks/elixir/examples/execute_operator_task_batch.exs
alias KyuubikiSdk.Auth
alias KyuubikiSdk.ControlPlaneClient

case System.argv() do
  [batch_path] ->
    base_url = System.get_env("KYUUBIKI_BASE_URL", "http://127.0.0.1:4000")

    auth =
      case System.get_env("KYUUBIKI_TOKEN") do
        nil -> nil
        token -> Auth.access_token(token)
      end

    batch = batch_path |> File.read!() |> Jason.decode!()
    client = ControlPlaneClient.new(base_url, auth: auth)
    {:ok, result} = ControlPlaneClient.execute_operator_task_batch(client, batch)
    IO.puts(Jason.encode_to_iodata!(result, pretty: true))

  _ ->
    IO.puts(:stderr, "usage: mix run examples/execute_operator_task_batch.exs batch.json")
    System.halt(64)
end

File: sdks/python/examples/execute_operator_task_batch.py
from __future__ import annotations

import json
import os
import sys
from pathlib import Path

from kyuubiki_sdk import ControlPlaneClient, KyuubikiAuth


def main() -> None:
    if len(sys.argv) != 2:
        raise SystemExit("usage: python examples/execute_operator_task_batch.py batch.json")

    base_url = os.getenv("KYUUBIKI_BASE_URL", "http://127.0.0.1:4000")
    token = os.getenv("KYUUBIKI_TOKEN")
    auth = KyuubikiAuth.access_token(token) if token else None
    batch = json.loads(Path(sys.argv[1]).read_text(encoding="utf-8"))

    client = ControlPlaneClient(base_url, auth=auth)
    result = client.execute_operator_task_batch(batch)
    print(json.dumps(result, indent=2))


if __name__ == "__main__":
    main()
